src/dfmt/formatter.d

```diff
@@ -2148,13 +2148,18 @@ const pure @safe @nogc:
         return index < tokens.length && tokens[index].type == tokenType;
     }
 
-    bool onNextLine() @nogc nothrow pure @safe
+    bool onNextLine() @nogc pure @safe
     {
         import dfmt.editorconfig : OptionalBoolean;
+        import std.algorithm.searching : count;
+
+        if (config.dfmt_keep_line_breaks == OptionalBoolean.f || index <= 0)
+        {
+            return false;
+        }
+        const tokenEndLine = tokens[index - 1].line + tokens[index - 1].text.count('\n');
 
-        return config.dfmt_keep_line_breaks == OptionalBoolean.t
-            && index > 0
-            && tokens[index - 1].line < tokens[index].line;
+        return tokenEndLine < tokens[index].line;
```

Comment:
can you add a comment explaining what is being done in these 2 lines? `tokenEndLine` isn't the most intuitive variable name to interpret

Reply:
👍 very nice, explains the need for `\n` too. I'm usually more of a fan of renaming variables to act as a kind of documentation, but that's a bit hard to do here.

Reply:
Yeah, I couldn't think of a better name. If you have an idea, I'll rename it.

Reply:
how about previousTokenEndLineNo

Reply:
Done